dice_ml/explainer_interfaces/explainer_base.py

```diff
@@ -307,9 +307,13 @@ def feature_importance(self, query_instances, cf_examples_list=None,
             local_importance=local_importances,
             summary_importance=summary_importance)
 
+    def predict_fn(self, input_instance):
+        """prediction function"""
+        return self.model.get_output(input_instance)
+
     def predict_fn_for_sparsity(self, input_instance):
         """prediction function for sparsity correction"""
-        return self.model.get_output(input_instance)
+        return self.predict_fn(input_instance)
```

Comment:
@gaugup this line is leading to an error in the tests. The issue is that `predict_fn` is also defined in `dice_kd`, so when the object is `DiceKD` class, the `predict_fn` from DiceKD gets called. Instead, we want the `predict_fn` of ExplainerBase to be called. 

This is the reason a separate function for sparsity is needed. Can we roll it back to say `self.model.get_output` inside `predict_fn_for_sparsity`? Or do you think there's a more elegant way to do this?

Reply:
Thanks @amit-sharma  for pointing this out. I just debugged it. I now have the function predict_fn_for_sparsity() call get_output() directly. We should mandate certain contracts for predict_fn_for_sparsity() and predict_fn() if these should return probabilities or actual prediction for classification scenarios.